packages/form-js-viewer/src/render/components/FormField.js

```diff
@@ -6,41 +6,47 @@ import {
 import {
   FormContext,
   FormRenderContext
-} from './context';
+} from '../context';
 
 import {
   findData,
   findErrors,
   pathStringify
-} from '../util';
+} from '../../util';
 
 const noop = () => false;
 
 
-export default function FormElement(props) {
+export default function FormField(props) {
   const {
     path,
     field,
     onChange
   } = props;
 
+  const {
+    getService
+  } = useContext(FormContext);
+
+  const formFieldRegistry = getService('formFieldRegistry'),
+        formFields = getService('formFields'),
+        form = getService('form');
+
   const {
     data,
     errors,
-    fields,
-    properties,
-    getFieldRenderer
-  } = useContext(FormContext);
+    properties
+  } = form._getState();
```

Comment:
Use of private API? Works as a temporary solution. We should consider to expose these properties as a public API.

Reply:
The reason I marked this as private was that it's supposed to be used by other services but not by users of the library since the schema isn't exported when calling `_getState`. Of cource from the viewpoint of other services the API is considered public. I decided to leave it as it is for now. I thought about adding `getData`, `getErrors` etc. but there's already `FormEditor#getSchema` which actually exports the schema. We might have to rethink this API.